client/settings/settings-records.js

```diff
@@ -0,0 +1,129 @@
+  mouseConfineOn: false,
+  // Sound settings
+  musicOn: false,
+  musicVolume: -1,
+  soundOn: false,
+  soundVolume: -1,
+  unitSpeechOn: false,
+  unitAcknowledgementsOn: false,
+  backgroundSoundsOn: false,
+  buildingSoundsOn: false,
+  gameSubtitlesOn: false,
+  cinematicSubtitlesOn: false,
+  originalVoiceOversOn: false,
+  original1998CampaignOn: false,
+  // Video settings
+  displayMode: -1,
+  fpsLimitOn: false,
+  fpsLimit: -1,
+  sdGraphicsFilter: -1,
+  vsyncOn: false,
+  hdGraphicsOn: false,
+  environmentEffectsOn: false,
+  realTimeLightingOn: false,
+  smoothUnitTurningOn: false,
+  shadowStackingOn: false,
+  pillarboxOn: false,
+  // Gameplay settings
+  gameTimerOn: false,
+  colorCyclingOn: false,
+  unitPortraits: -1,
+  minimapPosition: false,
+  apmDisplayOn: false,
+  apmAlertOn: false,
+  apmAlertValue: -1,
+  apmAlertColorOn: false,
+  apmAlertSoundOn: false,
+})
+
+// Mapping of setting names between SB and SC:R, where the names used in SB are the keys, and the
+// names used in SC:R are the values. Used for converting both from SC:R -> SB settings (when
+// fetching), and SB -> SC:R settings (when saving).
+export const scrSettingMappings = new Map([
+  ['keyboardScrollSpeed', 'm_kscroll'],
+  ['mouseScrollSpeed', 'm_mscroll'],
+  ['mouseSensitivityOn', 'MouseUseSensitivity'],
+  ['mouseSensitivity', 'MouseSensitivity'],
+  ['mouseScalingOn', 'MouseScaling'],
+  ['hardwareCursorOn', 'MouseHardwareCursor'],
+  ['mouseConfineOn', 'MouseConfine'],
+  ['musicOn', 'MusicEnabled'],
+  ['musicVolume', 'music'],
+  ['soundOn', 'SfxEnabled'],
+  ['soundVolume', 'sfx'],
+  ['unitSpeechOn', 'unitspeech'],
+  ['unitAcknowledgementsOn', 'unitnoise'],
+  ['backgroundSoundsOn', 'SoundInBackground'],
+  ['buildingSoundsOn', 'bldgnoise'],
+  ['gameSubtitlesOn', 'trigtext'],
+  ['cinematicSubtitlesOn', 'cinematicSubtitlesEnabled'],
+  ['originalVoiceOversOn', 'originalUnitVO'],
+  ['original1998CampaignOn', 'originalCampaign'],
+  ['displayMode', 'WindowMode'],
+  ['fpsLimitOn', 'FPSLimitEnabled'],
+  ['fpsLimit', 'FPSLimit'],
+  ['sdGraphicsFilter', 'SDFilterMode'],
+  ['vsyncOn', 'VSync'],
+  ['hdGraphicsOn', 'HDPreferences'],
+  ['environmentEffectsOn', 'ShowFoliage'],
+  ['realTimeLightingOn', 'RealtimeLightingEnabled'],
+  ['smoothUnitTurningOn', 'UseHDRotation'],
+  ['shadowStackingOn', 'ShadowStacking'],
+  ['pillarboxOn', 'OriginalAspectRatio'],
+  ['gameTimerOn', 'GameTimer'],
+  ['colorCyclingOn', 'ColorCycle'],
+  ['unitPortraits', 'UnitPortraits'],
+  ['minimapPosition', 'consoleSplit'],
+  ['apmDisplayOn', 'apm_Showing'],
+  ['apmAlertOn', 'apm_AlertUser'],
+  ['apmAlertValue', 'apm_AlertValue'],
+  ['apmAlertColorOn', 'apm_AlertUseColor'],
+  ['apmAlertSoundOn', 'apm_AlertUseSound'],
+])
+
+export function fromScrToSb(scrSettings) {
+  const mappedScrSettings = Object.entries(scrSettings).reduce((acc, [name, value]) => {
+    const sbKeyName = scrSettingMappings.findKey(scrKeyName => scrKeyName === name)
```

Comment:
I would *really* recommend just building the inverse map once at launch, otherwise this conversion is N^2. Should be simple, just:
```js
const sbToScrMapping = scrSettingMappings.entrySeq().map(([key, value]) => [value, key]).toMap()
```